File: content/snippets/typescript/design_patterns_command_modern.ts
// Modern TypeScript Command using () => void closures
// No Command interface or class hierarchy - commands are plain callable closures.

class Device {
    turnOn():  void { console.log("Device is ON"); }
    turnOff(): void { console.log("Device is OFF"); }
}

const device = new Device();

const commands: Array<() => void> = [
    () => device.turnOn(),
    () => device.turnOff(),
];

commands.forEach(cmd => cmd());

File: content/snippets/go/design_patterns_command_modern.go
package main

import "fmt"

// Modern Go Command using function closures.
// No Command interface or struct types needed.

type Device struct{}

func (d Device) TurnOn()  { fmt.Println("Device is ON") }
func (d Device) TurnOff() { fmt.Println("Device is OFF") }

func main() {
	device := Device{}

	commands := []func(){
		func() { device.TurnOn() },
		func() { device.TurnOff() },
	}

	for _, cmd := range commands {
		cmd()
	}
}
